tools/mcms_proposal/encoding.go

```diff
@@ -0,0 +1,159 @@
+package mcms_proposal
```

Comment:
Hm potentially, I think all of the components that are chain-specific should eventually move out into their own chain family specific folders. Lets keep thinking about this for now because of the top of my head:
* simulation
* execution
* operation/metadata encoding 
 
are all going to be chain specific so we should think more about how we want to structure it. I don't think we need to do this in this PR, we can address it in a follow up PR